tests/test_vega.py

```diff
@@ -263,7 +256,776 @@ def test_fill_anchor_in_string(tmp_dir):
+                    "dvc_inferred_y_value": "0.05",
+                    "predicted_test": "0.05",
+                    "actual": "0.5",
+                    "filename": "test",
+                    "field": "predicted",
+                },
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.9",
+                    "predicted_test": "0.9",
+                    "actual": "0.9",
+                    "filename": "test",
+                    "field": "predicted",
+                },
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.9",
+                    "predicted_train": "0.9",
+                    "actual": "0.9",
+                    "filename": "train",
+                    "field": "predicted",
+                },
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.9",
+                    "predicted_train": "0.9",
+                    "actual": "0.9",
+                    "filename": "train",
+                    "field": "predicted",
+                },
+            ],
+            "dvc_inferred_y_value",
+            [
+                {"filename": "test", "field": "predicted_test"},
+                {"filename": "train", "field": "predicted_train"},
+            ],
+            ["rev", "predicted", "actual"],
+            {"field": "filename::field", "sort": []},
+            ["rev", "filename::field", "dvc_inferred_y_value"],
+            ["rev", "filename::field", "actual"],
+        ),
+    ),
+)
+def test_optional_anchors_confusion(
+    datapoints,
+    y,
+    anchors_y_definitions,
+    expected_dp_keys,
+    row_encoding,
+    group_by_y,
+    group_by_x,
+):  # pylint: disable=too-many-arguments
+    props = {
+        "template": "confusion",
+        "x": "actual",
+        "y": y,
+        "revs_with_datapoints": ["B"],
+        "anchors_y_definitions": anchors_y_definitions,
+    }
+
+    expected_datapoints = _get_expected_datapoints(datapoints, expected_dp_keys)
+
+    renderer = VegaRenderer(datapoints, "foo", **props)
+    plot_content = renderer.get_filled_template()
+
+    assert plot_content["data"]["values"] == expected_datapoints
+    assert plot_content["facet"]["row"] == row_encoding
+    assert plot_content["spec"]["transform"][0]["groupby"] == [y, "actual"]
+    assert plot_content["spec"]["transform"][1]["groupby"] == group_by_y
+    assert plot_content["spec"]["transform"][2]["groupby"] == group_by_x
+    assert plot_content["spec"]["layer"][0]["width"] == 300
+    assert plot_content["spec"]["layer"][0]["height"] == 300
+
+
+@pytest.mark.parametrize(
+    ",".join(
+        [
+            "datapoints",
+            "y",
+            "anchors_y_definitions",
+            "expected_dp_keys",
+            "shape_encoding",
+            "tooltip_encoding",
+        ]
+    ),
+    (
+        (
+            [
+                {
+                    "rev": "B",
```

Comment:
Why start with changes to rev here when it isn't tested at all in the templates above?